### PR TITLE
[WIP]: Python3 RPC test runner

### PR DESCRIPTION
**Pull request by mdr0id**
Continued work from Py2 test runner and bash clean up for newer CI/CD models across platforms

### CODE REVIEW

File: qa/rpc-tests/test.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    total_stime = time.time()
+    for file in test_list:
+        logging.info("--- Running test : %s ---", file)
+        stats.append(file)
+        try:
+            start_time = time.time()
+            p1 = Popen(['python3', file])
+            buffer = p1.communicate()[0]
+        except:
+            fail_list.append(file)
```
Comment (daira) [inline qa/rpc-tests/test.py:167]: In the exception case, `p1` is not initialized, but it is used below.

**Comment by daira** [comment]
Upstream's current test runner is https://github.com/bitcoin/bitcoin/blob/master/test/functional/test_runner.py .

**Comment by mdr0id** [comment]
> Upstream's current test runner is https://github.com/bitcoin/bitcoin/blob/master/test/functional/test_runner.py . It has support for getting coverage data during tests.

Yes this one has made a lot of progress and even supports parallel RPCs, IIRC. Perhaps it is worth taking another look now that we are over the Py2 to Py3 migration.

File: qa/rpc-tests/test.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+    os.environ["BITCOIND"] = os.path.join(WORK_DIR, "Documents", "zcash", "zcashd")
+    os.environ["BITCOINCLI"] = os.path.join(WORK_DIR, "Documents", "zcash", "zcash-cli")
+    WORK_DIR = os.path.join(WORK_DIR,"zcash")
+elif HOST_OS == 'Linux':
+    os.environ["BITCOIND"] = os.path.join(WORK_DIR,"zcash", "zcashd")
+    os.environ["BITCOINCLI"] = os.path.join(WORK_DIR, "zcash", "zcash-cli")
+    WORK_DIR = os.path.join(WORK_DIR,"zcash")
+elif HOST_OS == 'Darwin':
+    os.environ["BITCOIND"] = os.path.join(WORK_DIR,"zcash", "zcashd")
+    os.environ["BITCOINCLI"] = os.path.join(WORK_DIR, "zcash", "zcash-cli")
```
Comment (rex4539) [inline qa/rpc-tests/test.py:43]:
```suggestion
    os.environ["BITCOIND"] = os.path.join(WORK_DIR,"zcash", "src", "zcashd")
    os.environ["BITCOINCLI"] = os.path.join(WORK_DIR, "zcash", "src", "zcash-cli")
```

File: qa/rpc-tests/test.py [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+WORK_DIR = os.path.expanduser('~')
+
+#@TODO Change these to defaults per platform of cwd
+if HOST_OS == 'Windows':
+    os.environ["BITCOIND"] = os.path.join(WORK_DIR, "Documents", "zcash", "zcashd")
+    os.environ["BITCOINCLI"] = os.path.join(WORK_DIR, "Documents", "zcash", "zcash-cli")
+    WORK_DIR = os.path.join(WORK_DIR,"zcash")
+elif HOST_OS == 'Linux':
+    os.environ["BITCOIND"] = os.path.join(WORK_DIR,"zcash", "zcashd")
+    os.environ["BITCOINCLI"] = os.path.join(WORK_DIR, "zcash", "zcash-cli")
```
Comment (rex4539) [inline qa/rpc-tests/test.py:39]:
```suggestion
    os.environ["BITCOIND"] = os.path.join(WORK_DIR,"zcash", "src", "zcashd")
    os.environ["BITCOINCLI"] = os.path.join(WORK_DIR, "zcash", "src", "zcash-cli")
```

**Comment by str4d** [comment]
Superceded by https://github.com/zcash/zcash/pull/4874.

**Pull request closed without merging**